main.go

```diff
@@ -222,3 +228,20 @@ func processEvents(ctx context.Context, logger hclog.Logger, extensionClient *ex
 		}
 	}
 }
+
+// createUserAgentFunc returns a function that can generate a useragent string that will be attached to the request param.
+// This function has access to the config state of the program via closure.
+func createUserAgentFunc(_ *api.Config, _ *config.AuthConfig) func(request *api.Request) string {
+	return func(request *api.Request) string {
+		// the DevEx team uses UAs that look like "vault-client-go/0.0.1 (Darwin arm64; Go go1.19.2); writing to temp; requesting via proxy"
+		buf := bytes.NewBufferString(fmt.Sprintf("%s/%s (%s %s; Go %s)", extensionName, extensionVersion, runtime.GOOS, runtime.GOARCH, runtime.Version()))
+		if sec, err := config.ParseConfiguredSecrets(); err != nil && len(sec) > 0 {
+			buf.WriteString("; writing to temp file")
+		}
+		if request.URL.Host == "127.0.0.1" {
```

Comment:
I'm not entirely sold on this check.

My understanding is that whether we should check the proxy or not is based on the cache settings set on both the configuration and the inbound request from a client, which I'm not sure how to get access to.